main.py

```diff
@@ -68,27 +67,50 @@ def do_create_room(self, arg):
     def do_add_person(self,arg):
         """Usage: add_person <person_fname> <person_lname> <FELLOW/STAFF> [<wants_accommodation>]"""
         check = Dojo().add_person(arg)
-        if check == True:
-                Dojo().assign_room(arg)
+        if check:
+                Dojo().allocate_room(arg)
```

Comment:
`Dojo().allocate_room` ???

Everytime you do `Dojo()`, you create a new instance of the class. 

You shouldn't do this.
It's bad design.

Create one instance instead 

```python
the_only_dojo_you_need = Dojo()
...
the_only_dojo_you_need.allocate_room(arg)
```

If you MUST access the instance method this way (i.e using the `Dojo().do_something format`), then you should implement the Dojo class using the singleton pattern 
(I challenge you to try this on the side just for fun 💪 )